cmd/kubeless/cmdUtils/cmd_utils.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright (c) 2016-2017 Bitnami
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmdUtils
+
+import (
+	"fmt"
+
+	"github.com/kubeless/kubeless/pkg/utils"
+	"github.com/spf13/pflag"
+)
+
+// ValidateDeploymentInputs validates that the combination of flags are valid for a deployment
+func ValidateDeploymentInputs(flags *pflag.FlagSet) error {
```

Comment:
The up-front validation is to avoid users to have to check the *controller* logs looking for errors. If you execute `kubeless function deploy ...` with a wrong set of flags the program exits without any error and without prompting anything which makes it difficult for new users to realize what is happening.
I see this like validating a "schema" for the users input, I don't think we can duplicate "real" logic in the client side, let me know if you have a suggestion for doing so.

Reply:
I finally removed this function, it is not going to cover all the possibilities and it is going to be difficult to maintain, we should probably retrieve the result of a deployment from the controller.